src/Physics/Hadronization/LeptoHadronization.cxx

```diff
@@ -0,0 +1,515 @@
+  assert(np>0);
+  TClonesArray * particle_list = new TClonesArray("TMCParticle", np);
+  particle_list->SetOwner(true);
+
+  bool isTop = false;
+  register unsigned int i = 0;
+  TMCParticle * particle = 0;
+  TIter particle_iter(pythia_particles);
+  while( (particle = (TMCParticle *) particle_iter.Next()) ) {
+    
+    // Final state particles can not be quarks or diquarks but colorless 
+    if(particle->GetKS() == 1) {
+      if( pdg::IsQuark(particle->GetKF()) || pdg::IsDiQuark(particle->GetKF()) ) {
+        LOG("LeptoHad", pERROR) << "Hadronization failed! Bare quark/di-quarks appear in final state!";
+        particle_list->Delete();
+        delete particle_list;
+        return 0;            
+      }  
+    }
+
+    // When top quark is produced, it is immidiately decay before hadronization. Then the decayed
+    // products are hadronized with the hadron remnants. Therefore, we remove the top quark from 
+    // the list of particles so that the mother/daugher assigments is at the same level for decayed
+    // products and hadron remnants.
+    if ( i==0 && pdg::IsTQuark( TMath::Abs(particle->GetKF()) ) ) { isTop=true; continue; }
+
+    // fix numbering scheme used for mother/daughter assignments
+    if ( isTop ) {
+      (particle->GetParent()==0) ? particle->SetParent(particle->GetParent() - 1) : particle->SetParent(particle->GetParent() - 2);
+      particle->SetFirstChild (particle->GetFirstChild() - 2);
+      particle->SetLastChild  (particle->GetLastChild()  - 2);
+    }
+    else  {
+      particle->SetParent(particle->GetParent() - 1);
+      particle->SetFirstChild (particle->GetFirstChild() - 1);
+      particle->SetLastChild  (particle->GetLastChild()  - 1);
+    }                                  
+    // insert the particle in the list
+    new ( (*particle_list)[i++] ) TMCParticle(*particle);
+  }
+
+  delete particle;
+  pythia_particles->Clear("C");
+
+  return particle_list;
+
+}
+//____________________________________________________________________________
+void LeptoHadronization::Configure(const Registry & config)
+{
+  Algorithm::Configure(config);
+  this->LoadConfig();
+}
+//____________________________________________________________________________
+void LeptoHadronization::Configure(string config)
+{
+  Algorithm::Configure(config);
+  this->LoadConfig();
+}
+//____________________________________________________________________________
+void LeptoHadronization::LoadConfig(void)
+{
+
+
+  GetParam("MaxIter-Had", fMaxIterHad ) ;
+
+  // Width of Gaussian distribution for transverse momentums
+  // Define in LEPTO with PARL(3) and PARL(14)
+  GetParam("Primordial-kT", fPrimordialKT ) ;
+  GetParam("Remnant-pT",    fRemnantPT ) ;
+
+  // It is, with quark masses added, used to define the minimum allowable energy of a colour-singlet parton system.
+  GetParam( "Energy-Singlet", fMinESinglet ) ; 
+
+  // PYTHIA parameters only valid for HEDIS
+  GetParam( "Wmin", fWmin ) ;
```

Comment:
Same comment on Wmin than in the GLRES. Can we make its name more specific? Is it the same quantity for GLRES? is it a different one?